Add Drift Detection Docs and Update Workspaces Notifications Page Content (#55)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/settings/notifications.mdx

# Run Notifications

> Terraform Cloud can use webhooks to notify external systems about the progress of runs. Each workspace has its own notification settings, and can notify up to 20 destinations.

Reviewer: Small nitpick: we should probably adjust the wording of the title and overview here to include workspace events since notifications are no longer a run-specific thing!

Author: ohh yeah!! that's a good point!! let me do that

Configuring notifications requires admin access to the workspace. Refer to [Permissions](/cloud-docs/users-teams-organizations/permissions) for details.

[permissions-citation]: #intentionally-unused---keep-for-maintainers

-> **API:** Refer to [Notification Configuration APIs](/cloud-docs/api-docs/notification-configurations).

## Viewing and Managing Notification Settings

> To add, edit, or delete notifications for a workspace, go to the workspace and Click **Settings > Notifications**. The **Notifications** page appears, showing existing notification configurations.

Author: ```suggestion
To add, edit, or delete notifications for a workspace, go to the workspace and click **Settings > Notifications**. The **Notifications** page appears, showing existing notification configurations.
```

## Creating a Notification Configuration

A notification configuration specifies a destination URL, a payload type, and the specific run events that should generate a notification. To create a notification configuration:

1. Go to the workspace's notifications settings and click **Create a Notification**. The **Create a Notification** form appears.
1. Complete the following fields to configure your notifications:

> 

Reviewer 2: can we parallelize these a bit better?

Author: i tried! see what you think

>    - **Destination:** Terraform Cloud can deliver either a generic payload, a payload formatted specifically for Slack, Microsoft Teams, or an Email. Choose which type you want from the drop-down. Refer to [Notification Payloads](#notification-payloads) for details.

Reviewer 2: ```suggestion
   - **Destination:** Terraform Cloud can deliver either a generic payload, or a payload formatted specifically for Slack, Microsoft Teams, or an Email. Choose which type you want from the drop-down. Refer to [Notification Payloads](#notification-payloads) for details.
```

   - **Name:** A display name to use for this notification configuration.

>    - **Webhook URL** Only available for generic, Slack, and Microsoft Teams webhooks. The destination URL for the webhook payload. This URL must accept HTTP or HTTPS `POST` requests and should be able to do something useful with the chosen payload type.

Author: ```suggestion
   - **Webhook URL** Only available for generic, Slack, and Microsoft Teams webhooks. The destination URL for the webhook payload. This URL must accept HTTP or HTTPS `POST` requests and should be able to use the chosen payload type.
```

     - For Slack notifications, refer to Slack's documentation on [how to create an incoming webhook](https://api.slack.com/messaging/webhooks#create_a_webhook).
     - For Microsoft Teams notifications, refer to Microsoft's documentation on [how to create an incoming webhook](https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook).
   - **Token** (Optional) Only available for generic webhooks. An arbitrary secret string that Terraform Cloud will use to sign its notification webhooks. Refer to [Notification Authenticity][inpage-hmac] for details. This setting is encrypted for storage, so you cannot view it after saving.

   - **Email Recipients** Only available for emails. Select the users from the workspace's organization to notify.

   - **Workspace Events:** You can notify this destination for all events or only for specific events. The following events are available.

>        - **Drift**: Every time Terraform Cloud detects configuration drift. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.

Reviewer 2: ```suggestion
       - **Drift**: Terraform Cloud detected configuration drift. Only available [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) is enabled for the workspace.
```

>        - **Drift Check Fail**: Every time a drift detection check fails. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.

Reviewer 2: ```suggestion
       - **Drift Check Fail**:  A drift detection check failed. Only available if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) is enabled for the workspace.
```

>    - **Run Events:** You can notify this destination for all events or only for specific events. The following events are available.

Reviewer 2: ```suggestion
   - **Run Events:** You can notify this destination for all events or only for specific events. The following events are available:
```

>    - **Workspace Events:** Whether to notify this destination for all events, or only for selected events. The following specific events are available.
>        - **Drift**: Every time Terraform Cloud detects configuration drift. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.
>        - **Drift Check Fail**: Every time a drift detection check fails. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.
>    - **Run Events:** Whether to notify this destination for all events, or only for selected events. The following specific events are available.

Author: ```suggestion
   - **Workspace Events:** You can notify this destination for all events or only for specific events. The following events are available.
       - **Drift**: Every time Terraform Cloud detects configuration drift. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.
       - **Drift Check Fail**: Every time a drift detection check fails. Only if [Drift Detection](/cloud-docs/workspaces/settings/drift-detection) has been enabled for the workspace.
   - **Run Events:** You can notify this destination for all events or only for specific events. The following events are available.
```
